CMakeLists.txt

```diff
@@ -40,8 +43,62 @@ if(MLIR_ENABLE_BINDINGS_PYTHON)
   message(STATUS "Enabling Python API")
   add_subdirectory(python)
 endif()
-add_subdirectory(test)
-add_subdirectory(quiccir-opt)
+
 add_subdirectory(quiccir-translate)
-add_subdirectory(quiccir-miniapp)
-add_subdirectory(external)
\ No newline at end of file
+add_subdirectory(quiccir-lib)
+
+option(QUICCIR_BUILD_TEST "Build quiccir tests." ON)
```

Comment:
Should we separate options and stuff into a separate cmake file in order to keep the top level CMakeLists.txt cleaner?

Reply:
yes, I cleaned up a bit the top level CMakeLists.txt